friprosveta/views.py

```diff
@@ -478,6 +479,44 @@ def allocations_json(request, timetable_slug=None):
     filtered_allocations = _allocation_set(param_ids,
                                            tt.allocations,
                                            request.user.is_staff)
+
+    #returns more information about allocations
+    request_r = request.GET
+    if 'mode' in request_r:
+        mode_l = request_r['mode']
+        if mode_l == 'ext':
+            multiple_allocations = namedtuple('AllocationVM', ['object', 'day_index', 'hour_index', 'duration'])
+            weekday_mapping = {wd[0]: i for i, wd in enumerate(WEEKDAYS)}
+            hour_mapping = {wh[0]: i for i, wh in enumerate(WORKHOURS)}
+            allocation_vms = [multiple_allocations(
+                object=allocation,
+                day_index=weekday_mapping[allocation.day],
+                hour_index=hour_mapping[allocation.start],
+                duration=allocation.duration,
+            ) for allocation in filtered_allocations]
+
+            allocation_vms = sorted(allocation_vms, key=lambda avm: avm.day_index)
+            allocations_by_day = [(d, list(avm_grouper))
+                                  for d, avm_grouper in itertools.groupby(allocation_vms, lambda avm: avm.object.day)]
+            allocations_ext = dict() 
+            for day,subjects in allocations_by_day:
```

Comment:
The other component is NOT a subject. A subject has its own meaning.
Please change name to `allocations` since they will actually be objects of type allocation.

Reply:
Changed to allocations.